include/highfive/H5DataType.hpp

```diff
@@ -136,6 +136,37 @@ class CompoundType : public DataType {
     void create(size_t size = 0);
 };
 
+template <typename T>
+class EnumType : public DataType {
+public:
+    struct member_def {
+        member_def(std::string t_name, T t_value)
+            : name(std::move(t_name))
+            , value(std::move(t_value)) {}
+        std::string name;
+        T value;
+    };
+
+    EnumType(const EnumType& other) = default;
+
+    EnumType(std::vector<member_def> t_members, DataType t_underlying_type)
+        : members(std::move(t_members))
+        , underlying_type(t_underlying_type) {
+        create();
+    }
+
+    EnumType(const std::initializer_list<member_def>& t_members, DataType t_underlying_type)
```

Comment:
can you use more specifically here `AtomicType<T>` instead of the more general `DataType` ?

Reply:
Fixed